build.gradle

```diff
@@ -21,12 +22,32 @@ reckon {
     stageFromProp('alpha', 'beta', 'rc', 'final')
 }
 
-sourceCompatibility = 11
+facets {
+    java17 {
+        parentSourceSet = 'main'
+    }
+}
+
+lombok {
+    version = "1.18.30"
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(11)
+    }
+    registerFeature('java17') {
```

Comment:
Lots of fancy stuff in this build.gradle to dual build java-11/17.
What is keeping us from raising the baseline to java 17 ?

Reply:
Backwards compatability with old gradle versions can imo be dropped:
Projects that are not updating the gradle wrapper (for whatever reason), would not be updating the enterprise-conventions-gradle either.

Reply:
It's not about compatibility with old Gradle versions; it is about compatibility with the JVM that is used for the Gradle build.
I believe a lot of Alfresco builds are still Java 11; requiring Java 17 requires work to update build JVMs everywhere (toolchains don't help for this)

Reply:
Reflecting back; we don't need any fancy Java 17 things.
Because we don't even need a dependency on the spring boot plugin, as we are always loading the class using reflection (because we can't access it directly; see below). This work is a leftover from when I was trying to get it to compile and run with Java code, which needed the spring boot plugin classes.